src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -288,7 +288,10 @@ private static void invokeWithTimeoutWithNewExecutor(ITestNGMethod tm, Object in
       ThreadTimeoutException exception = new ThreadTimeoutException("Method "
           + tm.getQualifiedName() + "()"
           + " didn't finish within the time-out " + realTimeOut);
-      exception.setStackTrace(exec.getStackTraces()[0]);
+      StackTraceElement[][] stacktraces = exec.getStackTraces();
+      if (stacktraces.length > 0) {

```

Comment:
Is it possible to have something different to 0?
Should we log something if the length != 0 because they are unexpected values?